src/openms/include/OpenMS/DATASTRUCTURES/QTCluster.h

```diff
@@ -119,115 +118,97 @@ namespace OpenMS
+     * 
+     * Has no functionality without a QTCluster (-head) pointing to it.
+     * Create object of this class before calling constructor of QTCluster
      */
-    double optimizeAnnotations_();
-
-public:
+    class OPENMS_DLLAPI Data_
+    {
+      /// QTCluster can access and manage all members of Data_
+      friend class QTCluster;
+
+      /// Pointer to the cluster center
+      GridFeature* center_point_;
+
+      /// unique id of this cluster
+      Size id_;
+
+      /**
+       * @brief Map that keeps track of the best current feature for each map
+       *
+       */
+      NeighborMap neighbors_;
+
+      /**
+       * @brief Temporary map tracking *all* neighbors
+       *
+       * For each input run, a multimap which contains pointers to all
+       * neighboring elements and the respective distance.
+       *
+       */
+      NeighborMapMulti tmp_neighbors_;
+
+      /// Maximum distance of a point that can still belong to the cluster
+      double max_distance_;
+
+      /// Number of input maps
+      Size num_maps_;
+
+      /// x coordinate in the grid cell
+      Int x_coord_;
+
+      /// y coordinate in the grid cell
+      Int y_coord_;
+
+      /**
+       * @brief Set of annotations of the cluster
+       *
+       * The set of peptide sequences that is compatible to the cluster center
+       * and results in the best cluster quality.
+       */
+      std::set<AASequence> annotations_;
+    };
 
     /**
      * @brief Detailed constructor
+     * @param data Pointer to internal data
      * @param center_point Pointer to the center point
      * @param num_maps Number of input maps
      * @param max_distance Maximum allowed distance of two points
      * @param use_IDs Use peptide annotations?
+     * @param id Unique ID of this cluster
      */
-    QTCluster(GridFeature* center_point, Size num_maps,
-              double max_distance, bool use_IDs, 
-              Int x_coord, Int y_coord);
 
-    /// Destructor
-    virtual ~QTCluster();
+    QTCluster(Data_* data, GridFeature* center_point, 
+              Size num_maps, double max_distance, 
+              bool use_IDs, Int x_coord, Int y_coord,
+              Size id);
+
+    /// Default constructor not accessible
+    QTCluster() = delete;
+
+    /// use this only when you actually want to move, but an interface can only copy
+    QTCluster(const QTCluster& rhs) = default;
+
+    /// cheap move because the bulk data lies in the Data_ member
+    QTCluster(QTCluster&& rhs) = default;
+
+    /// use this only when you actually want to move, but an interface can only copy
+    QTCluster& operator=(const QTCluster& rhs) = default;
+
+    /// cheap move because the bulk data lies in the Data_ member
+    QTCluster& operator=(QTCluster&& rhs) = default;
+
+    ~QTCluster() = default;
 
     /// Returns the cluster center
-    GridFeature* getCenterPoint();
+    GridFeature* getCenterPoint() const;
```

Comment:
can this be `const GridFeature* const`? (or at lease one of them?).

Reply:
The second const doesn't make sense because we return by value (compiler told us). But we added the first const and a lot of other consts in the whole code for this algorithm.